src/cargo/util/command_prelude.rs

```diff
@@ -440,8 +444,20 @@ pub trait ArgMatchesExt {
         self.maybe_flag("keep-going")
     }
 
-    fn targets(&self) -> Vec<String> {
-        self._values_of("target")
+    fn targets(&self) -> CargoResult<Vec<String>> {
+        if self.is_present_with_zero_values("target") {
+            let cmd = if is_rustup() {
+                "rustup target list"
+            } else {
+                "ustc --print target-list"
```

Comment:
Not a real objection. Just this recommendation could often be false when using `+channel` command line override.

Any typo

```suggestion
                "rustc --print target-list"
```

Reply:
Typo fixed.  I can go either way in how we handle the recommendation.